src/EncryptionOracle.sol

```diff
@@ -16,6 +16,15 @@ struct Ciphertext {
     DleqProof dleq;
 }
 
+/// @notice Struct that Medusa nodes submits in response to a request
+struct ReencryptedCipher {
+    G1Point random;
+    // TODO: Note this is not strictly useful and can be removed to decrease
```

Comment:
Oh I didn't realize that. So given the original cipher + a new random point, a user with a matching private key can decrypt the original cipher?

Reply:
Yes, the original cipher has two parts: C1 and C2. C2 stays fixed all the time, it doesn't really need to be onchain tbh even, only C1, uint256, is changing with reencryption. 
In my PoC initially I bundled the two but we can be more gas efficient I guess by removing C2 everywhere in the picture. That being said, for example in Holonym case, where we want to validate the ciphertext is correctly constructed from meaningful data, C2 is needed, but not at the Medusa level.

Reply:
Ok, I thought C2 was needed for the DLEQ proof. And I see that C2 can be helpful for applications to do additional checks.

Let's leave both C2 and the duplicate cipher for now and maybe do a gas efficiency sprint before official mainnet release.